proposals/lip-0057.md

```diff
@@ -1343,23 +1343,25 @@ def getActiveValidators(validatorsTwoRoundsAgo: list[EligibleValidatorObject], r
         nbrInitValidators = initRounds + NUMBER_ACTIVE_VALIDATORS - roundNumber
         nbrElectedValidators = NUMBER_ACTIVE_VALIDATORS - nbrInitValidators
 
-        # In the definition below, recall that validatorsTwoRoundsAgo is sorted by validator weight.
-        electedValidatorAddresses = [item.address for item in validatorsTwoRoundsAgo[:nbrElectedValidators]]
-
-        remainingInitValidators = [address for address in initValidators if address not in electedValidatorAddresses]
-
-        # Concatenation of elected validators and remaining initial validators.
-        activeValidatorAddresses = electedValidatorAddresses + remainingInitValidators[:nbrInitValidators]
-        activeValidators = [{"address": item, "bftWeight": 1 } for item in activeValidatorAddresses]
-
+        # Recall that validatorsTwoRoundsAgo is sorted by validator weight.
+        electedValidators = validatorsTwoRoundsAgo[:nbrElectedValidators]
+        electedValidatorAddresses = [item.address for item in electedValidators]
+
+        chosenInitValidators = [address for address in initValidators if address not in electedValidatorAddresses][:nbrInitValidators]   
+        
+        # Determine active validators and their BFT weight.
+        activeValidators = [{"address": item.address, "bftWeight": ceiling(item.weight, WEIGHT_SCALE_FACTOR) } for item in electedValidators]
+        averageElectedValidatorWeight = sum([item.weight for item in electedValidators]) // nbrElectedValidators
```

Comment:
I was first confused and thought this definition uses `activeValidators` instead of `electedValidators` and was then wondering why we divide by `WEIGHT_SCALE_FACTOR` again. 

Therefore optional suggestion:
```
activeValidators = [{"address": item.address, "bftWeight": ceiling(item.weight, WEIGHT_SCALE_FACTOR) } for item in validatorsTwoRoundsAgo[:nbrElectedValidators]]
averageElectedValidatorWeight = sum([item.weight for item in activeValidators]) // nbrElectedValidators
activeValidators+= [{"address": item, "bftWeight": averageElectedValidatorWeight } for item in chosenInitValidators]
```
and not define `electedValidators`.

Reply:
this looks cleaner to me as well

Reply:
LGTM

Reply:
Indeed, the initial version had a different order but with current status your proposed refactoring makes sense.

I used `electedValidators` to avoid slicing `validatorsTwoRoundsAgo` multiple times but I checked the implementation and it is implemented in a careful efficient way so I removed it as you suggested (in fact even `electedValidatorsAddresses` is not used in the implementation and could be removed, but I just keep it for better readability)